File: go_project/leetcode/lt119.go
package main

import "fmt"

func getFac(n int64) int64 {
	if n == 0 {
		return 1
	}
	ans := int64(1)
	for i := int64(2); i <= n; i++ {
		ans *= i
	}
	return ans
}

func getCom(n int, m int) int {
	x := getFac(int64(m))
	fmt.Println(x)
	y := getFac(int64(n))
	z := getFac(int64(m - n))
	return int(x / (y * z))
}

func getRow(rowIndex int) []int {
	f := make([]int, 0)
	for i := 0; i <= rowIndex; i ++ {
		f = append(f, getCom(i, rowIndex))
	}
	return f
}

func main() {
	n := 21
	fmt.Println(getRow(n))
}

File: python_project/lt119.py
from typing import List


class Solution:
    def getFac(self, n: int) -> int:
        if n == 0:
            return 1
        ans = 1
        for i in range(2, n + 1):
            ans *= i
        return ans

    def getCom(self, n: int, m: int) -> int:
        x = self.getFac(m)
        y = self.getFac(n)
        z = self.getFac(m - n)
        return x // (y * z)

    def getRow(self, rowIndex: int) -> List[int]:
        ans = []
        for i in range(0, rowIndex + 1):
            ans.append(self.getCom(i, rowIndex))
        return ans
